File: ts/project/.sdk/src/cmp/kotlin/fragment/Entity.fragment.kt
package KOTLINPACKAGE.entity

import KOTLINPACKAGE.core.Context
import KOTLINPACKAGE.core.Entity
import KOTLINPACKAGE.core.Helpers
import KOTLINPACKAGE.core.SdkClient
import KOTLINPACKAGE.utility.struct.Struct

/** EntityName entity client for the ProjectName SDK. */
@Suppress("UNCHECKED_CAST", "UNUSED_PARAMETER", "UNUSED_VARIABLE")
class EntyClass(clientIn: SdkClient, entoptsIn: MutableMap<String, Any?>?) :
  EntityBase("entityname", clientIn, entoptsIn) {

  override fun make(): Entity {
    val opts = LinkedHashMap(this.entopts)
    return EntyClass(this.client, opts)
  }

// #LoadOp

// #ListOp

// #CreateOp

// #UpdateOp

// #RemoveOp
}

File: ts/project/.sdk/src/cmp/elixir/fragment/Entity.fragment.ex
# ProjectName SDK EntityName entity
#
# Per-entity module. Generic construction/data/match operations delegate to
# EntityBase; each active op (load/list/create/update/remove) builds a ctx
# and drives it through ProjectName.Pipeline.run_op.

defmodule ProjectName.Entity.EntityName do
  alias Voxgig.Struct, as: S
  alias ProjectName.Helpers, as: H
  alias ProjectName.{EntityBase, Context, Pipeline}

  def new(client, entopts \\ nil) do
    EntityBase.construct(__MODULE__, client, "entityname", entopts)
  end

  def get_name(ent), do: EntityBase.get_name(ent)
  def make(ent), do: EntityBase.make(ent)
  def data_set(ent, args \\ nil), do: EntityBase.data_set(ent, args)
  def data_get(ent), do: EntityBase.data_get(ent)
  def match_set(ent, args \\ nil), do: EntityBase.match_set(ent, args)
  def match_get(ent), do: EntityBase.match_get(ent)

  # Streaming operation (see EntityBase.stream): runs `action` through the
  # pipeline and returns a lazy Stream over result items.
  def stream(ent, action, args \\ nil, callopts \\ nil),
    do: EntityBase.stream(ent, action, args, callopts)

  # #LoadOp

  # #ListOp

  # #CreateOp

  # #UpdateOp

  # #RemoveOp
end
